EDD Bookings Migrations

mecha opened the pull request:
~Resolves #2~

----

Migrations have been [moved to a different module](https://github.com/RebelCode/rcmod-wp-bookings-cqrs/pull/3). This PR now makes the module compliant with the migration changes made in `rcmod-wp-bookings-cqrs` and changes the DB version option name to be specific to EDD Bookings.

mecha committed:
Installed wp core and CQRS packages

mecha committed:
Added migrator class

mecha committed:
Added migrator service

mecha committed:
Moved migration file and added down migration

mecha committed:
Moved migrations under a version number directory

mecha committed:
Fixed module construction and added events

mecha committed:
Migrator now uses a more generic mechanism

mecha committed:
Updated module class to properly use new migrator

mecha committed:
Updated config format

mecha committed:
Added migrations config

mecha committed:
Fixed config name for migrations config

mecha committed:
Changed to using `mysqli` for executing queries

Also now recording the updated db version on successful migration

mecha committed:
Fixed some problems in first migration SQL

XedinUnknown commented on config/config.php:
```diff
@@ -6,11 +6,22 @@
- */
-$cfg['cqrs']['table_prefix'] = '${wpdb_prefix}eddbk_';
+return [
+    'cqrs' => [
```
If this is top-level configuration, then it needs to be contained under a node specific to the module.

mecha replied:
Changing this will break the many things that depend on it. This should be done as a separate task and done carefully, across all concerned modules. There is a card on this [project](https://github.com/orgs/RebelCode/projects/2#card-10257746).

XedinUnknown commented on migrations/1/up.sql:
```diff
@@ -20,18 +9,17 @@ create table wp_eddbk_bookings
+);
 
-create table wp_eddbk_session_rules
+create table if not exists ${table_prefix}session_rules
```
IMO the table names should come from config.

XedinUnknown commented on src/EddBkCqrsModule.php:
```diff
@@ -42,7 +66,18 @@ public function setup()
-            []
+            [
+                'eddbk_mysqli'   => function (ContainerInterface $c) {
+                    return new mysqli(DB_HOST, DB_USER, DB_PASSWORD, DB_NAME);
```
More coupling to environment, and even worse - to WP in particular. We should work out a way to remove usage of global constants from modules.

mecha replied:
Agree. Any recommended alternative?

mecha committed:
Flattened migrations directort contents

mecha committed:
Flattened migrations directory contents

Also replaced tokens to be the full table names.

mecha commented:
Migrations have been [moved to a different module](https://github.com/RebelCode/rcmod-wp-bookings-cqrs/pull/3).

mecha committed:
Removed migrations and added config override

This module now simply changes the db version option name to be specific to EDDBK.

mecha committed:
Merge branch 'task/2_simple-migration' of https://github.com/rebelcode/rcmod-eddbk-cqrs into task/2_simple-migration

mecha committed:
Merge branch 'develop' into task/2_simple-migration

mecha committed:
Updated changelog
